code-challenges.test.js

```diff
@@ -23,30 +38,98 @@ const hitchhikersCharacters = [
+    //going to declare a variable called lastNameUpperCase which will store the first letter of the first name  by accessing with charAt(0), and concat the rest of the substring from character 1. Going to repeat process for lastname.
+    //declare a varibale called addOcupation which will have the value of ocupation
+    //going to string intorpolate in order to receive a sentence and return that
+    //lastly return firstLastNameCapital array from the function
+
 
 // --------------------2) Create a function that takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.
 
 // a) Create a test with an expect statement using the variables provided.
 
+describe("mixedDataArray", () => {
+  it("returns an array of only the REMAINDERS of the numbers when divided by 3.", () => {
+    const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
+      // Expected output: [ 2, 0, -1, 0 ]
+   expect(mixedDataArray(hodgepodge1)).toEqual([ 2, 0, -1, 0 ])
+   const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
+   // Expected output: [ 2, 1, -1 ]
+   expect(mixedDataArray(hodgepodge2)).toEqual([ 2, 1, -1 ])
+  })
+})
+
 const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
 // Expected output: [ 2, 0, -1, 0 ]
 const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
 // Expected output: [ 2, 1, -1 ]
 
 // b) Create the function that makes the test pass.
 
+const mixedDataArray = (param) => {
+  return param.filter(value => typeof value  === "number" )
+  .map(value => value % 3)
+}
+
 // Pseudo code:
+      //going to create function called mixedDataArray that will take one parameter
+      //going to filter over the parameter by using .filter() to return a new array of filtered values
+      //going to identify numbers only by using typeof operator 
+      //going to use .map() to iterate through the numbers and return the remainders by using the modulo operator when divided by 3
+
 
 // --------------------3) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
 
 // a) Create a test with an expect statement using the variables provided.
 
+describe("arrOfNum", () => {
+  it("returns the sum of all the numbers cubed.", () => {
+    const cubeAndSum1 = [2, 3, 4]
+    // Expected output: 99
+   expect(arrOfNum(cubeAndSum1)).toEqual(99)
+   const cubeAndSum2 = [0, 5, 10]
+   // Expected output: 1125
+   expect(arrOfNum(cubeAndSum2)).toEqual(1125)
+  })
+})
+
+        // ReferenceError: arrOfNum is not defined
+
 const cubeAndSum1 = [2, 3, 4]
 // Expected output: 99
 const cubeAndSum2 = [0, 5, 10]
 // Expected output: 1125
 
 // b) Create the function that makes the test pass.
 
+const arrOfNum = (param) => {
+  let initialValue = 0
+  let cubedOfNums = param.map(value => value **3)
+  return cubedOfNums.reduce((acc, currentValue) =>
+    acc + currentValue, initialValue
```

Comment:
I would remove initialValue and just put 0 here to set the starting value of the accumulator. This way there's no need to create a variable on line 123 and saves a line of code